### DISCUSSION TITLE
Generic x86_64 build? - Qotom-Q555G6-S05

### DISCUSSION THREAD

**Discussion by CroutonCrusader** [SmoothWAN/SmoothWAN#4] [General & Help]
Hi -- I happened across your Reddit post and hate that you were banned, because this project looks very promising and helpful. I'm unsure how else to contact you, so I'm opening this up as an "issue".

I'm currently running Ubuntu Desktop with Speedify client installed on a Qotom-Q555G6-S05 x86 mini PC.

I'd like to try out the x86 build. Will you be posting an x86 build?

**Comment by TalalMash** [comment]
Hi! 
Done, I'd appreciate a small feedback if it works well with the Qotom, particularly the ethernet ports, interesting hardware.

**Comment by CroutonCrusader** [comment]
I loaded smoothwan-0.92-x86-64-generic-ext4-combined-efi.img to a USB drive and booted the Qotom. This is where it stops during the boot process
![TinyPilot-2021-12-27T19_55_44 224Z](https://user-images.githubusercontent.com/96717851/147503366-db6fcbe1-eba9-4fd3-9387-c17c3ff349e3.jpg)
.

**Comment by TalalMash** [comment]
If pressing enter gets you to the console then its working properly, the Web UI is now accessible at 192.168.3.1 from ethernet. 
E1: I just checked kmod-igb for that switch (i211) in build configuration, it's enabled. If you can't access the WebUI type "ip a" in console there should be "br-lan".

**Comment by CroutonCrusader** [comment]
I pressed enter and it gave me a console. I have eth0 connected to my T-Mobile gateway, eth4 connected to my switch, and eth5 connected to a Peplink router. My Peplink router is reporting "No cable connected" and when I type "ip a", it lists all ports as "DOWN"

When I try to access 192.168.3.1 using a PC connected to the switch, there doesn't appear to be anything there.

![TinyPilot-2021-12-27T20_54_26 244Z](https://user-images.githubusercontent.com/96717851/147506778-335f5752-c7bb-41e6-9741-32c214a9781b.jpg)
.

**Comment by TalalMash** [comment]
Hmm, root@(none) suggests that there is mounting issue, can you use the non EFI image and switch to BIOS? I just tested EFI with 2 UEFI systems and it worked well, maybe the image is corrupted or problematic flashdrive/storage.

**Comment by TalalMash** [comment]
Moved to discussions as it seems to be an OpenWRT specific issue with Qotom if I'm not mistaken. I have tested UEFI with others remotely on: Yanling Atom, MSI H310M. Please try using regular OpenWRT https://downloads.openwrt.org/releases/21.02.1/targets/x86/64/openwrt-21.02.1-x86-64-generic-ext4-combined-efi.img.gz to confirm. Also try using a different flash drive.

**Comment by CroutonCrusader** [comment]
Sorry for delay in responding -- work has kept my hands tied.

I will try another USB drive. I will also try the other .img file

I will report back.

**Comment by nmz787** [comment]
What are the requirements to run this build? Anything specific? I have a few CherryTrail CPU based systems that would be pretty power efficient, if performant enough. I also have other older machines that are basically just sitting in storage that could be booted up for basic testing, and to (possibly) wait out the chip shortage for a cheaper raspi 4

**Reply by TalalMash** [reply]
> What are the requirements to run this build? Anything specific? I have a few CherryTrail CPU based systems that would be pretty power efficient, if performant enough. I also have other older machines that are basically just sitting in storage that could be booted up for basic testing, and to (possibly) wait out the chip shortage for a cheaper raspi 4

Any recent x86 machine from 2012 and up are better than a Pi, particurarly AES acceleration which Speediy switches to when available.
There are no specifics other than the need to disable Secure Boot in boot manager menu if you use UEFI images.
You can use a USB flash drive instead of using the internal storage.
To use the internal, flash Ubuntu to a USB key, navigate to the browser and download SmoothWAN x86, UEFI or BIOS depending on your machine, then run balenaEtcher and toggle `Show hidden` and select the internal drive, after finish, reboot, unplug the flash drive and you're all set. If you have multiple ethernet adapters, only one of them is LAN.
